adding AVIATAR user story blog post (#449)
quarkusio/quarkusio.github.io: _posts/2020-02-14-aviatar-experiences-significant-savings.adoc

=== The Service Discovery service

The Service Discovery service is used to allow automatic discovery between the microservices AVIATAR consist of. It is considered high risk because if it breaks down, it would have a major impact on customers. Also, the original version of this service ran in production as a highly available service in an application server, consuming a lot of cloud resources. The Quarkus version of this service has been running in native mode in Development for about 3 months with no problems and on January 18, 2020, it was deployed to production to replace the instance running on the application server. It should also be mentioned that this Quarkus service started in JVM mode because it was using MongoDB and there was no MongoDB client Quarkus extension when its development started. But as soon as the MongoDB client Quarkus extension became available, they were able to switch the entire service to native mode. This speaks to the fast innovation and new contributions that are part of the Quarkus open source community project.

== Why not Spring Boot?

> Lufthansa Technik see Quarkus as a great fit in their journey to running their services in a serverless mode. They have many services that aren’t invoked that often but they still need to have 3 instances of each up and running all the time for high-availability requirements leading to high cloud resource consumption costs. They plan to turn these seldom accessed services to Function-as-a-service calls so that they can be invoked on demand leading to a reduction of cloud. If they were using Spring Boot, it would take too long to boot up, making it prohibitive to use in a serverless mode.

Reviewer: ```suggestion
Lufthansa Technik sees Quarkus as a great fit in their journey to running their services in a serverless mode. They have many services that aren’t invoked that often but they still need to have 3 instances of each up and running all the time for high-availability requirements leading to high cloud resource consumption costs. They plan to turn these seldom accessed services to Function-as-a-service calls so that they can be invoked on demand leading to a reduction of cloud. If they were using Spring Boot, it would take too long to boot up, making it prohibitive to use in a serverless mode.
```

> Likewise, they have experienced lower memory and compute cloud resource consumption when using Quarkus and according to Thorsten, _“with Quarkus, they could run 3 times denser deployments without sacrificing availability and response times of services.”_

Reviewer 2: I would suggest to put something like the following:

"With Quarkus and its use of GraalVM, they could run"

as the denser deployments come from the combination of the two technologies.

Author: Modifying as follows since I can't really change the approved quote from Lufthansa Technik:

Likewise, they have experienced lower memory and compute cloud resource consumption when using Quarkus - plus its use of GraalVM - and according to Thorsten, “with Quarkus, they could run 3 times denser deployments without sacrificing availability and response times of services.”

> Their developers are Spring Boot developers with Java EE experience as well, so the learning curve for Quarkus was very small since its syntax and approach was _“close to what our developers are already doing and it’s familiar to them. This is a big benefit,”_ Thorsten affirmed.

Reviewer: ```suggestion
Their developers are Spring Boot developers with Java EE experience as well, so the learning curve for Quarkus was very small since its syntax and approach was _“close to what our developers are already doing and it’s familiar to them. This is a big benefit”_, Thorsten affirmed.
```

> With respect to the recently introduced Spring API compatibility feature in Quarkus, Thorsten said that _“they may use the Spring API compatibility in Quarkus when migration their current Spring Boot microservices to Quarkus. However, for developing new microservices, they plan to use just the Quarkus APIs directly because it would be awkward to use another API within Quarkus.”_

Reviewer: ```suggestion
With respect to the recently introduced Spring API compatibility feature in Quarkus, Thorsten said that _“they may use the Spring API compatibility in Quarkus when migrating their current Spring Boot microservices to Quarkus. However, for developing new microservices, they plan to use just the Quarkus APIs directly because it would be awkward to use another API within Quarkus.”_
```

== The Quarkus Benefits

> Among the many benefits that Quarkus provides to Lufthansa Technik, Thorsten mentioned that _“they could cut cloud resource costs threefold."_ And the same goes with OpenShift because of the higher density you can achieve on each core using Quarkus. A tangible example is for the one mentioned earlier for an existing microservice that was consuming ½ core plus 1 GB of RAM per instance, which required 1.5 cores and 3 GB of RAM when running it in HA on the cloud (3 instances). When using the Quarkus version of the same microservice, its consumption was of 200 milicores plus 200-400 MB of RAM per instance. This translates to 0.6 cores plus 600 MB – 1.2 GB of RAM for an HA deployment of 3 instanced of the microservice. _“They could run 3 times denser deployments without sacrificing availability and response times of services,”_ Thornsten reiterated.

Reviewer: `A tangible example is for the one mentioned earlier for an existing microservice` <- this looks weird to me.

Author: Rephrased. Fixed.

> Among the many benefits that Quarkus provides to Lufthansa Technik, Thorsten mentioned that _“they could cut cloud resource costs threefold."_ And the same goes with OpenShift because of the higher density you can achieve on each core using Quarkus. A tangible example is for the one mentioned earlier for an existing microservice that was consuming ½ core plus 1 GB of RAM per instance, which required 1.5 cores and 3 GB of RAM when running it in HA on the cloud (3 instances). When using the Quarkus version of the same microservice, its consumption was of 200 milicores plus 200-400 MB of RAM per instance. This translates to 0.6 cores plus 600 MB – 1.2 GB of RAM for an HA deployment of 3 instanced of the microservice. _“They could run 3 times denser deployments without sacrificing availability and response times of services,”_ Thornsten reiterated.

Reviewer: ```suggestion
Among the many benefits that Quarkus provides to Lufthansa Technik, Thorsten mentioned that _“they could cut cloud resource costs threefold."_ And the same goes with OpenShift because of the higher density you can achieve on each core using Quarkus. A tangible example is for the one mentioned earlier for an existing microservice that was consuming ½ core plus 1 GB of RAM per instance, which required 1.5 cores and 3 GB of RAM when running it in HA on the cloud (3 instances). When using the Quarkus version of the same microservice, its consumption was of 200 millicores plus 200-400 MB of RAM per instance. This translates to 0.6 cores plus 600 MB – 1.2 GB of RAM for an HA deployment of 3 instances of the microservice _“. They could run 3 times denser deployments without sacrificing availability and response times of services”_, Thornsten reiterated.
```

> Among the many benefits that Quarkus provides to Lufthansa Technik, Thorsten mentioned that _“they could cut cloud resource costs threefold."_ And the same goes with OpenShift because of the higher density you can achieve on each core using Quarkus. A tangible example is for the one mentioned earlier for an existing microservice that was consuming ½ core plus 1 GB of RAM per instance, which required 1.5 cores and 3 GB of RAM when running it in HA on the cloud (3 instances). When using the Quarkus version of the same microservice, its consumption was of 200 milicores plus 200-400 MB of RAM per instance. This translates to 0.6 cores plus 600 MB – 1.2 GB of RAM for an HA deployment of 3 instanced of the microservice. _“They could run 3 times denser deployments without sacrificing availability and response times of services,”_ Thornsten reiterated.

Reviewer 2: lets do add a mention of GraalVM native-image here.

Author: Appended at the end of the paragraph, the following sentence:

These are the types of optimizations that can only be achieved by the symbiotic combination of Quarkus and GraalVM.

> Thorsten also described Quarkus live coding capabilities as a _“really good thing.”_ Many of their applications have web-based user interfaces and _“making changes and reloading pages instantaneously is a great feature,”_ Thorsten affirmed.

Reviewer: ```suggestion
Thorsten also described Quarkus live coding capabilities as a _“really good thing”_. Many of their applications have web-based user interfaces and _“making changes and reloading pages instantaneously is a great feature”_, Thorsten affirmed.
```

> Another benefit, already mentioned earlier, was the small Quarkus learning curve experienced by their developers, who are Spring Boot developers with Java EE experience. What makes this possible is the stack of technologies included in Quarkus, composed of best-of-breed and familiar technologies for Kubernetes-native microservices. Some of the Quarkus extensions used by the AVIATAR developers are: Java Web Token (JWT), JAX-RS, Mongo DB client, MicroProfile Rest Client, Keycloak (security), MicroProfile Metrics, Hibernate ORM (Relational DB), and MicroProfile Health Check.

Reviewer: ```suggestion
Another benefit, already mentioned earlier, was the small Quarkus learning curve experienced by their developers, who are Spring Boot developers with Java EE experience. What makes this possible is the stack of technologies included in Quarkus, composed of best-of-breed and familiar technologies for Kubernetes-native microservices. Some of the Quarkus extensions used by the AVIATAR developers are: Java Web Token (JWT), JAX-RS, MongoDB client, MicroProfile Rest Client, Keycloak (security), Hibernate ORM (for relational databases), MicroProfile Metrics, and MicroProfile Health Check.
```
